arch/arm/core/aarch64/mmu/arm_mmu.c

```diff
@@ -433,7 +440,8 @@ static int arm_mmu_init(const struct device *arg)
 	__asm__ volatile("mrs %0, sctlr_el1" : "=r" (val));
 	__ASSERT((val & SCTLR_M) == 0, "MMU is already enabled\n");
 
-	setup_page_tables(&kernel_ptables);
+	if (!MPIDR_TO_CORE(GET_MPIDR()))
```

Comment:
Is Zephyr assuming that core 0 will always be the core that starts first? If so, ok, although I'd prefer that that was documented somewhere.

Reply:
Currently yes. But I tend to make zephyr could starts from other cores in future. I'll add a comment here.